### PR TITLE
BB-905: Add Legacy Address interface

### PR DESCRIPTION
**Pull request by stanis64**
This change is [Reviewable](https://reviewable.io/reviews/bit-oasis/coin/9)

### CODE REVIEW

**Comment by foowie** [comment]
Please add tests

**Comment by stanis64** [comment]
Tests added

File: tests/unit/BitOasis/Coin/Address/LitecoinAddressTest.php [hunk 1]
```diff
@@ -59,4 +59,14 @@ protected static function getCurrency() {
...
 	}
 
+	/**
+	 * @param string $legacyFormat
+	 * @param string $newFormat
+	 * @dataProvider providerToNewAddressFormat
+	 */
+	public function testLegacyAddress($legacyFormat, $newFormat) {
+		$litecoinAddress = new LitecoinAddress($newFormat, self::getCurrency());
+		$this->assertEquals($litecoinAddress->getLegacyAddress(), $legacyFormat);
```
Comment (foowie) [inline tests/unit/BitOasis/Coin/Address/LitecoinAddressTest.php]: `Lciocvp1PvQyMH9Srmxn7dSC94Gk2YtCLm` is not legacy format of `Lciocvp1PvQyMH9Srmxn7dSC94Gk2YtCLm` address. `Lciocvp1PvQyMH9Srmxn7dSC94Gk2YtCLm` has no legacy format.

Comment (foowie) [reply tests/unit/BitOasis/Coin/Address/LitecoinAddressTest.php]: In that case it should be returning null.

File: tests/unit/BitOasis/Coin/Address/LitecoinAddressTest.php [hunk 2]
```diff
@@ -39,7 +39,10 @@ public function testToNewAddressFormat($legacyFormat, $expectedFormat) {
 	 */
 	public function testToLegacyAddressFormat($expectedFormat, $newFormat) {
 		$address = new LitecoinAddress($newFormat, self::getCurrency());
-		$this->assertEquals($address->toLegacyAddressFormat()->toString(), $expectedFormat);
+		$legacyAddress = $address->toLegacyAddressFormat();
+		if ($legacyAddress !== null) {
+			$this->assertEquals($legacyAddress->toString(), $expectedFormat);
```
Comment (foowie) [inline tests/unit/BitOasis/Coin/Address/LitecoinAddressTest.php]: This is not how tests should work. What if `$legacyAddress` is null but `$expectedFormat` is not?

File: src/BitOasis/Coin/Address/LitecoinAddress.php [hunk 3]
```diff
@@ -121,7 +121,7 @@ public function toLegacyAddressFormat() {
 			return new static(Base58Check::encodeAddress($decodedAddress), $this->currency);
 		}
 		
-		return $this;
+		return null;
```
Comment (foowie) [inline src/BitOasis/Coin/Address/LitecoinAddress.php]: You should not change current API of library. Or have you checked all projects if it won't do any bc break?

**Pull request merged**